src/main/java/com/pkb/unit/Unit.java

```diff
@@ -106,15 +99,15 @@ public Unit(String id, Bus bus, long retryPeriod, TimeUnit retryTimeUnit) {
                 .subscribe(ignored -> handleReportDependencies());
 
         Observable.interval(retryPeriod, retryTimeUnit, Schedulers.computation())
-                .subscribe(this::handleRetry);
+                .subscribe(ignored -> handleRetry());
 
         // Advertise our full current state
         unchecked(() -> bus.sink().accept(message(NewUnit.class).withPayload(newUnit(id))));
         handleReportDependencies();
         handleReportState();
     }
 
-    private void handleRetry(Long ignored) {
+    private void handleRetry() {
```

Comment:
We could consider subscribe and unsubscribe when the Unit is made Unset so that we don't call this method unnecessarily. At the same time I think a maxTrialCount could be also introduced maybe not to try to restart a Unit infinitely.

Reply:
Good spot, that could be an optimization. 
re. timeout: maybe, but why would I want to stop retrying?

Reply:
Naturally this is not urgent.

Reply:
I just thought that if a Unit could not be started 100 times in a row then most probably it will never start but I don't insist on it :)